### PR TITLE
Feature/tri 290 swagger documentation irs

### PR DESCRIPTION
**Pull request by mkanal**
Thank you in advance.

### CODE REVIEW

**Review by ds-jkreutzfeld — CHANGES_REQUESTED** [review]
I fixed the checkstyle issues to have a clean build.
The API comparison test is now failing, please check whether this test is still relevant.

File: irs-api/src/main/java/net/catenax/irs/exceptions/IrsApiException.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import net.catenax.irs.component.enums.StatusCodeEnum;
+
+/**
+ * Irs API request exception
+ */
+@Schema(description = "Detailed exception information")
+@Value
+@Jacksonized
+@Builder(toBuilder = true)
+public class IrsApiException {
```
[the remark this answers is not in the document]
Comment (mkanal) [reply irs-api/src/main/java/net/catenax/irs/exceptions/IrsApiException.java:28]: Will be removed. Is not used.

File: irs-api/src/main/java/net/catenax/irs/requests/IrsPartsTreeRequest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * IrsPartsTree request object
+ */
+@Value
+public class IrsPartsTreeRequest extends IrsPartsTreeRequestBase {
+
+    @Pattern(regexp = IrsApiConstants.GLOBAL_ASSET_ID_REGEX,
+            message = ApiErrorsConstants.NOT_BLANK)
+    @NotBlank
+    @Size(min = 25, max = 25)
```
[the remark this answers is not in the document]
Comment (mkanal) [reply irs-api/src/main/java/net/catenax/irs/requests/IrsPartsTreeRequest.java]: @Size(min = IrsApiConstants.JOB_ID_SIZE, max = IrsApiConstants.JOB_ID_SIZE)

**Review by jzbmw — APPROVED** [review]
Ive overlooked some topics. Im by far not qualified to check if these topics are valid or not

**Pull request merged**